src/tools/clippy/clippy_lints/src/int_plus_one.rs

```diff
@@ -52,8 +53,8 @@ enum Side {
 
 impl IntPlusOne {
     #[expect(clippy::cast_sign_loss)]
-    fn check_lit(lit: &Lit, target_value: i128) -> bool {
-        if let LitKind::Int(value, ..) = lit.kind {
+    fn check_lit(token_lit: token::Lit, target_value: i128) -> bool {
```

Comment:
I agree that the naming could be overhauled, and it likely makes sense for `lit`/`Lit` to be for token-level literals and then `MetaItemLit` or similar to be used elsewhere. However, I would prefer to do that in a follow-up, because this PR is taking a long time and I think a proper naming overhaul will turn out to be quite a large change, covering `ast::Lit`, `ast::LitKind`, and probably other types as well.

In the meantime, I have found the name `token_lit` very helpful to make it clear which kind of literal is being used.